src/ProjectNeon/Assets/Data/Templates/TurnAI.cs

```diff
@@ -2,5 +2,5 @@
 
 public abstract class TurnAI : ScriptableObject
 {
-    public abstract PlayedCard Play();
+    public abstract PlayedCard Play(Enemy activeEnemy);
```

Comment:
So, I like this idea, but it's not possible if we have AIs as Scriptable Objects. 

ScriptableObjects are permanent singletons. So, if we connected `HunterWeakOpponentAI` to an enemy, no other enemy anywhere could use that AI. And it would impact future games, too. 

On the other hand, if we don't use Scriptable Objects, then Designers can't set Enemy Behaviors in the editor.

So, it's better not to bind the Enemy to the AI, and use function parameter injection instead.